src/main/java/uk/gov/hmcts/reform/hmc/service/common/OverrideAuditService.java

```diff
@@ -0,0 +1,161 @@
+
+    private final HearingStatusAuditRepository hearingStatusAuditRepository;
+    private final LinkedHearingStatusAuditRepository linkedHearingStatusAuditRepository;
+    private final UrlManager roleAssignmentUrlManager;
+    private final UrlManager dataStoreUrlManager;
+
+    // To overcome circular dependency issue
+    @Lazy @Autowired @Setter
+    private SecurityUtils securityUtils;
+
+    public void logOverrideAudit(HttpServletRequest request) {
+        String roleAssignmentUrl = request.getHeader(roleAssignmentUrlManager.getUrlHeaderName());
+        String dataStoreUrl = request.getHeader(dataStoreUrlManager.getUrlHeaderName());
+
+        if (!isLogRequired(roleAssignmentUrl, dataStoreUrl)) {
+            return;
+        }
+
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode root = mapper.createObjectNode();
+
+        String path = request.getRequestURI();
+        root.put("path", path);
+
+        if (roleAssignmentUrl != null && !roleAssignmentUrlManager.getHost().equals(roleAssignmentUrl)) {
+            root.put("role-assignment-url", roleAssignmentUrl);
+        }
+
+        if (dataStoreUrl != null && !dataStoreUrlManager.getHost().equals(dataStoreUrl)) {
+            root.put("data-store-url", dataStoreUrl);
+        }
+
+        root.put("hmctsServiceName", getServiceName(request.getHeader(SERVICE_AUTHORIZATION)));
+
+        if (isRequestWithBody(request)) {
+            root.put("requestBody", getRequestBody(request));
+        }
+
+        root.put("requestTimestamp", LocalDateTime.now(ZoneId.of("UTC")).toString());
+        root.put("user-id", securityUtils.getUserId());
+
+        if (path.startsWith("/linkedHearingGroup")) {
+            String groupId = getAttributeId(request, "id");
+            saveLinkedHearingStatusAudit(groupId, root);
+        } else {
+            String hearingId = getAttributeId(request, "id");
+            saveHearingStatusAudit(hearingId, root);
+        }
+    }
+
+    private void saveLinkedHearingStatusAudit(String groupId, ObjectNode root) {
+        LinkedHearingStatusAuditEntity auditEntity = new LinkedHearingStatusAuditEntity();
+        auditEntity.setHmctsServiceId("n/a");
+        auditEntity.setLinkedGroupId(groupId);
+        auditEntity.setLinkedGroupVersion("n/a");
+        auditEntity.setLinkedHearingEvent("n/a");
+        auditEntity.setOtherInfo(root);
+        linkedHearingStatusAuditRepository.save(auditEntity);
+    }
+
+    private void saveHearingStatusAudit(String hearingId, ObjectNode root) {
+        HearingStatusAuditEntity auditEntity = new HearingStatusAuditEntity();
+        auditEntity.setHearingId(hearingId);
+        auditEntity.setHmctsServiceId("n/a");
+        auditEntity.setStatus(OVERRIDE_URL);
+        auditEntity.setHearingEvent(OVERRIDE_URL_EVENT);
+        auditEntity.setRequestVersion("n/a");
+        auditEntity.setOtherInfo(root);
+        hearingStatusAuditRepository.save(auditEntity);
```

Comment:
can we have source and target set please

Reply:
Addressed.